daemon/k8s_watcher.go

```diff
@@ -1364,6 +1364,9 @@ func (d *Daemon) addCiliumNetworkPolicyV2(ciliumV2Store cache.Store, cnp *cilium
 					return errors.New("Received object of unknown type from API server, expecting v2.CiliumNetworkPolicy")
```

Comment:
Can't you move this whole code to the `updateRuleAnnotations` function?

Reply:
I could factor it out? I don't see why I'd move it out of this existing function; if `updateRuleAnnotations` is executed, this function is not hit.

Reply:
@aanm factored out into a separate function for use in `updateRuleAnnotations` and the update of the CNP when the spec / specs have changed.